packages/vite-plugin-checker/src/checkers/eslint/main.ts

```diff
@@ -63,7 +65,20 @@ const createDiagnostic: CreateDiagnostic<'eslint'> = (pluginConfig) => {
         ...translatedOptions,
         ...pluginConfig.eslint.dev?.overrideConfig,
       }
-      const eslint = new ESLint(eslintOptions)
+
+      let eslint: ESLint
+      if (pluginConfig.eslint.useFlatConfig) {
```

Comment:
Is it necessary to add this field here? 🤔 As per the documentation of [eslint](https://eslint.org/docs/latest/use/configure/migration-guide#:~:text=To%20use%20flat%20config%20with,ESLINT_USE_FLAT_CONFIG%20environment%20variable%20to%20true%20.), user could control enabling flat config by `eslint.config.js` or `ESLINT_USE_FLAT_CONFIG`. The only usage for this field I can figure out is force using legacy config when multiple config files exists together.

Reply:
I want it to be consistent with the vscode eslint setting `eslint.experimental.useFlatConfig` for the time being, because this was not the default configuration file before eslint 9.0. And it seems that eslint versions before 8.50 do not support flat config, so there is no `eslint/use-at-your-own-risk`.

Reply:
I just checker vscode-eslint plugin and I believe this's a smart choice for the time being until flatConfig turned out to be the default config type.